dali/test/python/test_pipeline.py

```diff
@@ -1654,3 +1654,102 @@ def test_invoke_serialize_error_handling_string():
 @raises(TypeError, "*define_graph*callable*")
 def test_invoke_serialize_error_handling_not_string():
     _identity_pipe().serialize(42)
+
+
+def check_dtype_ndim(dali_pipeline, output_dtype, output_ndim):
+    def ndim_dtype_matches(test_value, ref_value):
+        ref_value = ref_value if type(ref_value) is list else [ref_value]
+        for tv, ref in zip(test_value, ref_value):
+            if ref is None:
+                if tv != -1:
+                    return False
+            else:
+                if tv != ref:
+                    return False
+        return True
```

Comment:
I guess this is good example to support @klecki proposal to get rid of -1.

Reply:
Done